src/index.js

```diff
@@ -1,86 +1,110 @@
-  self.app.post(`${self.baseApi}/all`, notificationController.updateAllUnread);
+    // Expose endpoints
+    self.app.get(`${self.baseApi}/all/:userid`, notificationController.getAll);
+    self.app.put(`${self.baseApi}/:notificationId`, notificationController.updateUnread);
+    self.app.delete(`${self.baseApi}/:notificationId`, notificationController.delete);
+    self.app.post(`${self.baseApi}/all`, notificationController.updateAllUnread);
 
-  self.io.on('connection', (socket) => {
-    socket.emit('who-are-you');
+    self.io.on('connection', (socket) => {
+      socket.emit('who-are-you');
 
-    socket.on('check-in', (user, ack) => {
-      self.io.sockets[user.id] = socket;
-      self.sendUnread(user.id);
+      socket.on('check-in', (user, ack) => {
+        self.io.sockets[user.id] = socket;
+        self.sendUnread(user.id);
 
-      // Send to client endpoint base route forexposed endpoints
-      ack(`{"baseApi":"${self.baseApi}"}`);
+        // Send to client endpoint base route forexposed endpoints
+        ack(`{"baseApi":"${self.baseApi}"}`);
+      });
+
+      socket.on('check-news', (user) => {
+        self.sendUnsent(user.id);
+      });
     });
+  }
 
-    socket.on('check-news', (user) => {
-      self.sendUnsent(user.id);
+  /**
+   * Send notification using socket
+   * @param  {string} event        custom socket event
+   * @param  {object} notification object representation for notification
+   */
+  send(event, notification) {
+    const user = notification.to;
+    this.io.sockets[user].emit(event, notification, (ack) => {
+      if (ack) {
+        notificationController
+        .update(notification._id, { sent: true }, //eslint-disable-line
+          (err) => {
+            if (err) throw err;
+          });
+      }
     });
-  });
-};
+  }
 
-Notification.prototype.send = (event, notification) => {
-  const user = notification.to;
-  this.io.sockets[user].emit(event, notification, (ack) => {
-    if (ack) {
+  /**
+   * Sends all the unread notifications by user id
+   * @param  {string} userId user identifier
+   */
+  sendUnread(userId) {
+    const self = this;
+    if (userId) {
       notificationController
-      .update(notification._id, { sent: true }, //eslint-disable-line
-        (err) => {
-          if (err) throw err;
+      .getUnread(userId, (err, data) => {
+        if (err) throw err;
+        data.forEach((notification) => {
+          self.send('unread', notification);
         });
+      });
     }
-  });
-};
+  }
 
-Notification.prototype.sendUnread = (userId) => {
-  const self = this;
-  if (userId) {
-    notificationController
-    .getUnread(userId, (err, data) => {
-      if (err) throw err;
-      data.forEach((notification) => {
-        self.send('unread', notification);
+  /**
+   * Sends all unsent notifications by user id-
+   * @param  {string} userId User identifier
+   */
+  sendUnsent(userId) {
+    const self = this;
+    if (userId) {
```

Comment:
Exists some possible use cases where this function is called and userId is not defined or null? I mean I think it's not responsability of this function check if is a not null or not undefined value. what do you thik @EddSuarez @jac1013